packages/tasit-identity-contract/src/GnosisSafe.test.js

```diff
@@ -0,0 +1,152 @@
+import Action from "tasit-action";
+const { ConfigLoader, ERC20 } = Action;
+const { DetailedERC20 } = ERC20;
+import { ethers } from "ethers";
+import { expect } from "chai";
+import GnosisSafe from "./GnosisSafe";
+import { local as localAddresses } from "../../tasit-contracts/3rd-parties/gnosis/addresses";
+import { createFromPrivateKey } from "../../tasit-account/dist/testHelpers/helpers";
+import {
+  mineBlocks,
+  createSnapshot,
+  revertFromSnapshot,
+  confirmBalances,
+} from "tasit-action/dist/testHelpers/helpers";
+
+// The goal of this integration test suite is to use only exposed classes
+// from TasitSdk. ProviderFactory is used here as an exception
+// as the clearest way to get a provider
+// in this test suite. Eventually, maybe ProviderFactory may move to
+// some shared helper dir.
+import ProviderFactory from "tasit-action/dist/ProviderFactory";
+
+const { GnosisSafe: GNOSIS_SAFE_ADDRESS } = localAddresses;
+const ERC20_ADDRESS = "0x37E1A58dD465D33263D00185D065Ee36DD34CDb4";
+
+const { utils: ethersUtils } = ethers;
+const { bigNumberify } = ethersUtils;
+
+const ZERO = 0;
+const ONE = bigNumberify(`${1e18}`);
+
+// 100 gwei
+const GAS_PRICE = bigNumberify(`${1e11}`);
+
+describe("GnosisSafe", () => {
+  const config = {
+    provider: {
+      network: "other",
+      provider: "jsonrpc",
+      pollingInterval: 50,
+      jsonRpc: {
+        url: "http://localhost",
+        port: 8545,
+      },
+    },
+    events: {
+      timeout: 2000,
+    },
+  };
+  let gnosisSafeContract;
+  let gnosisSafeUtils;
+  let anaWallet;
+  let snapshotId;
+  let provider;
+  let erc20;
+
+  before("", async () => {
+    ConfigLoader.setConfig(config);
+
+    provider = ProviderFactory.getProvider();
+
+    const anaPrivateKey =
+      "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60";
+
+    anaWallet = createFromPrivateKey(anaPrivateKey);
+
+    // Contract deployment setup with ana (accounts[0]) as the only owner
+    // To change that, edit the file "tasit-contract/3rd-parties/gnosis/scripts/2_deploy_contracts.js"
+    gnosisSafeContract = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+    erc20 = new DetailedERC20(ERC20_ADDRESS);
+  });
+
+  beforeEach("", async () => {
+    snapshotId = await createSnapshot(provider);
+
+    const etherBalance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${etherBalance}`).to.equal(`${ZERO}`);
+
+    const erc20Balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${erc20Balance}`).to.equal(`${ZERO}`);
+
+    //gnosisSafeContract.removeWallet();
+    //erc20.removeWallet();
+  });
+
+  afterEach("", async () => {
+    await revertFromSnapshot(provider, snapshotId);
+  });
+
+  it("wallet owner should deposit and withdraw some ethers", async () => {
```

Comment:
While the owner is "withdrawing" from the contract in a sense, I might word it as the multisig contract sending ethers instead

Reply:
Sorry, I don't got it.

Reply:
In other words, the name of the test could be: `"wallet owner should deposit to contract account, and contract account should send ethers back to [owner/user]`" (I'm not sure if it will be sent to the owner or another user in a new version of the test.

Reply:
Hmmm. I've split the deposit/faucet to another block. This test case structure/names is now:

describe("test cases that needs ETH deposit to the wallet", async () => {
    beforeEach("faucet", async () => { ... });
    it("wallet owner should withdraw some ethers", async () => { ... });
});

Should I rename `wallet owner should withdraw some ethers` to `
contract account should send ethers back to owner` ?

Reply:
I saw that, thanks. Looks good!

Yep, I like that renaming

Reply:
Done.

Reply:
Great, thanks. Resolving this.